Sources/Customization/Protocols/OPTEventProcessor.swift

```diff
@@ -0,0 +1,34 @@
+/****************************************************************************
+* Copyright 2019, Optimizely, Inc. and contributors                        *
+*                                                                          *
+* Licensed under the Apache License, Version 2.0 (the "License");          *
+* you may not use this file except in compliance with the License.         *
+* You may obtain a copy of the License at                                  *
+*                                                                          *
+*    http://www.apache.org/licenses/LICENSE-2.0                            *
+*                                                                          *
+* Unless required by applicable law or agreed to in writing, software      *
+* distributed under the License is distributed on an "AS IS" BASIS,        *
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+* See the License for the specific language governing permissions and      *
+* limitations under the License.                                           *
+***************************************************************************/
+    
+
+import Foundation
+
+public protocol OPTEventProcessor {
+    
+    func process(event: BatchEvent, completionHandler: DispatchCompletionHandler?)
```

Comment:
We should be processing `UserEvent`s such as `ImpressionEvent` or `ConversionEvent`. Please take a look at these: 

https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/event/internal/UserEvent.java

https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/event/internal/ConversionEvent.java

The "batch" should be a batch created from `UserEvent`s

Reply:
I understand BatchEvent is a wrong name here, but we used it for a while. I thought about changing the name and create new UserEvent type, but we already have too much name/interface changes. Will look at possible changes.

Reply:
But this is a new protocol right? We shouldn't be worries about `BatchEvent` breaking a new protocol since no one is using it yet

Reply:
Yep, it's not about breaking changes. I'll fix it.

Reply:
whispering.... we can do a typealias to userevent = batch event.

Reply:
Not a bad idea, but I already made all the changes to support UserEvent :)